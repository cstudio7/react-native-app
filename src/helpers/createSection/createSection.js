import { groupBy } from 'lodash';

const createSection = names => {
  const groups = groupBy(names, nameObj => nameObj.name.substr(0, 1));
  return Object.keys(groups).map(groupName => ({
    title: groupName,
    data: groups[groupName]
  }));
};

export default createSection;
