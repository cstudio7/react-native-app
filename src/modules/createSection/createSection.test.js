import createSection from './createSection';

const mockedNames = Object.freeze([
  {
    name: 'Artur',
    isFavorite: false
  },
  {
    name: 'Alexander',
    isFavorite: false
  },
  {
    name: 'Svetlana',
    isFavorite: true
  }
]);

it('renders correctly', () => {
  expect(createSection(mockedNames)).toMatchSnapshot();
});
