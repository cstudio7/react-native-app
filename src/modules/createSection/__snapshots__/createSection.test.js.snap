// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  Object {
    "data": Array [
      Object {
        "isFavorite": false,
        "name": "Artur",
      },
      Object {
        "isFavorite": false,
        "name": "Alexander",
      },
    ],
    "title": "A",
  },
  Object {
    "data": Array [
      Object {
        "isFavorite": true,
        "name": "Svetlana",
      },
    ],
    "title": "S",
  },
]
`;
